src/api/catalog-search/catalog-search.ts

```diff
@@ -109,18 +114,42 @@ export class CatalogSearchAPI {
         (pkg) => !pkg.keywords?.includes("construct-hub/hide-from-search")
       )
       .reduce((map, pkg) => {
-        const { name, version } = pkg;
+        const { author, name, version } = pkg;
         const id = [name, version].join("@");
 
         const downloads = stats.packages[name]?.downloads?.npm ?? 0;
+        let [scope, packageName] = name.split("/");
+
+        if (!packageName) {
+          packageName = scope;
+        }
+
+        let authorName: string | undefined;
+        let authorEmail: string | undefined;
+
+        if (typeof author === "string") {
+          authorName = author;
+        } else {
+          if (author?.name) {
+            authorName = author.name;
+          }
 
+          if (author?.email) {
+            authorEmail = author.email;
+          }
+        }
```

Comment:
I've moved all this data mapping to when we define the catalog map, rather than when we pass it to the index to prevent any confusion / mutability related bugs